File: joao pedro/Problema0300/Problema0300.java
import java.util.Scanner;
public class Problema0300{
    public static void main(String[] args) {
        /*Faça um programa que leia um conjunto de números ordenados e armazene-o em um vetor X.
        Faça um algoritmo que verifique se um dado valor Y existe em X. Caso não exista, inclua Y em X 
        de forma que X continue ordenado. Obs.: Não deve ser usado nenhum processo de ordenação. */
        Scanner input=new Scanner(System.in);
        int n=0,check=0;
        
        double y;

        System.out.println("Digite a quantidade de numeros no vetor X:");
        n=Integer.valueOf(input.nextLine());
        double vetorX[]=new double[n];
        for(int i=0;i<n;i++){
            System.out.println("Digite um numero na posicao "+i+" do vetor X: ");
            vetorX[i]=Double.valueOf(input.nextLine());

        }
        System.out.println("Digite um valor Y: ");
        y=Double.valueOf(input.nextLine());

        for(int i=0;i<n;i++){
            if(y==vetorX[i]){
                System.out.println("O valor Y existe no vetor X");
                check=1;

            }
            
        }
        if(check==0){
            System.out.println("O valor Y nao esta no vetor X. Incluindo valor Y");
            vetorX[0]=y;
        }
        for(int i=0;i<n;i++){
            System.out.println("Vetor X na posicao "+i+": "+vetorX[i]);
        }
    }


}

File: joao pedro/Problema0300/Problema0300.js
const prompt = require("prompt-sync")({ sigint: true });

var vetorX=[],n=0,check=0,y;

n=Number(prompt("Digite o tamanho do vetor X: "));

for(i=0;i<n;i++){
    vetorX[i]=Number(prompt("Digite um numero na posicao "+i+" do vetor X: "));
    

}
y=Number(prompt("Digite um valor Y: "));


for(i=0;i<n;i++){
    if(y==vetorX[i]){
        console.log("O valor Y existe no vetor X");
        check=1;

    }
    
}
if(check==0){
    console.log("O valor Y nao esta no vetor X. Incluindo valor Y");
    vetorX[0]=y;
}
for(i=0;i<n;i++){
    console.log("Vetor X na posicao "+i+": "+vetorX[i]);
}